modules/ingest-common/src/main/java/org/opensearch/ingest/common/ConvertProcessor.java

```diff
@@ -118,6 +119,16 @@ public Object convert(Object value) {
                 return value.toString();
             }
         },
+        IP {
+            @Override
+            public Object convert(Object value) {
+                if (value instanceof String && InetAddresses.isInetAddress(value.toString())) {
+                    return value;
```

Comment:
This function validates the object value and doesn't convert it.  Should you be returning an InetAddress object?

Reply:
I've tried that at the first time, but found [error](https://github.com/opensearch-project/OpenSearch/blob/13604c88cf31a402e3c771ed4ec90466b594fa1b/libs/core/src/main/java/org/opensearch/core/xcontent/XContentBuilder.java#L863) is thrown when serializing,  and IpFieldType can [accept string value](https://github.com/opensearch-project/OpenSearch/blob/13604c88cf31a402e3c771ed4ec90466b594fa1b/server/src/main/java/org/opensearch/index/mapper/IpFieldMapper.java#L202), so it's simpler that we use the original string value instead of an InetAddress, the result are same.

Reply:
Please add that in the comments